src/scanner.c

```diff
@@ -51,6 +51,14 @@ static bool scan_whitespace_and_comments(TSLexer *lexer) {
   }
 }
 
+bool scan_for_word(TSLexer *lexer, char* word, unsigned len) {
+    advance(lexer);
+    for (unsigned i = 0; i < len; i++) {
```

Comment:
hmm what happens if the file contains less than len characters remaining? What does
advance(lexer) do when there is no more characters to read?

Reply:
Good question -> the first `advance(lexer)` in the function advances past the `e` character (or another character chosen) in the switch statement (so not a problem), and the `lexer->lookahead` check in the if statement catches whether the file ends after.

I tested this on a file with just the `e` and an empty file, and both of them parse correctly 👍

Reply:
perfect.